### PR TITLE
YC-604: Add type name in email subject

### PR DESCRIPTION
**Pull request by rbovard**
Fix https://jira.liip.ch/browse/YC-604

Please review.

### CODE REVIEW

**Comment by rbovard** [comment]
@monodo I fixed most of the test with adapting the mail subjects. But I don't understand why the remaining tests are failling, can you help me?

**Comment by rbovard** [comment]
@monodo As discussed, only displaying the unique type names in subject.

Could you please help me with the test?

**Comment by rbovard** [comment]
Thanks a lot @AlexandreJunod!

@monodo I've rebased the PR, could you please review and merge?

File: permits/services.py [hunk 1]
```diff
@@ -936,9 +936,20 @@ def submit_permit_request(permit_request, request):
...
     )
+
+    # Using a set to have only unique values
+    work_type_names_set = set()
+    for wot in permit_request.works_object_types.all():
+        work_type_names_set.add(str(wot.works_type))
+    work_type_names = ", ".join(
+        work_type_name for work_type_name in work_type_names_set
+    )
+
```
Comment (monodo) [inline permits/services.py:947]: In one line maybe ?
`works_object_types.all().values_list("work_type_name")`

File: permits/services.py [hunk 2]
```diff
@@ -1016,8 +1028,18 @@ def request_permit_request_validation(permit_request, departments, absolute_uri_
...
     )
 
+    # Using a set to have only unique values
+    work_type_names_set = set()
+    for wot in permit_request.works_object_types.all():
+        work_type_names_set.add(str(wot.works_type))
+    work_type_names = ", ".join(
+        work_type_name for work_type_name in work_type_names_set
+    )
+
```
Comment (monodo) [inline permits/services.py:1038]: Create a function in services.py for a DRYer approach ?

File: permits/services.py [hunk 3]
```diff
@@ -1045,8 +1067,19 @@ def send_validation_reminder(permit_request, absolute_uri_func):
...
     )
+
+    # Using a set to have only unique values
+    work_type_names_set = set()
+    for wot in permit_request.works_object_types.all():
+        work_type_names_set.add(str(wot.works_type))
+    work_type_names = ", ".join(
+        work_type_name for work_type_name in work_type_names_set
+    )
+
```
Comment (monodo) [inline permits/services.py:1078]: idem

File: permits/views.py [hunk 4]
```diff
@@ -456,11 +456,23 @@ def handle_amend_form_submission(self, form):
...
+            permit_request = form.instance
+
+            # Using a set to have only unique values
+            work_type_names_set = set()
+            for wot in permit_request.works_object_types.all():
+                work_type_names_set.add(str(wot.works_type))
+            work_type_names = ", ".join(
+                work_type_name for work_type_name in work_type_names_set
+            )
+
```
Comment (monodo) [inline permits/views.py:468]: idem, please use a DRYer approach

File: permits/views.py [hunk 5]
```diff
@@ -531,9 +543,21 @@ def handle_validation_form_submission(self, form):
...
                 ):
+
+                    # Using a set to have only unique values
+                    work_type_names_set = set()
+                    for wot in self.permit_request.works_object_types.all():
+                        work_type_names_set.add(str(wot.works_type))
+                    work_type_names = ", ".join(
+                        work_type_name for work_type_name in work_type_names_set
+                    )
+
```
Comment (monodo) [inline permits/views.py:554]: idem

File: permits/views.py [hunk 6]
```diff
@@ -1442,9 +1466,20 @@ def permit_request_classify(request, permit_request_id, approve):
...
             classify_form.save()
+
+            # Using a set to have only unique values
+            work_type_names_set = set()
+            for wot in permit_request.works_object_types.all():
+                work_type_names_set.add(str(wot.works_type))
+            work_type_names = ", ".join(
+                work_type_name for work_type_name in work_type_names_set
+            )
+
```
Comment (monodo) [inline permits/views.py:1477]: idem

**Review by monodo — CHANGES_REQUESTED** [review]

**Review by monodo — APPROVED** [review]

**Pull request merged**